### PR TITLE
fix: add animation capabilities to navigation, self user profile (AR-1101)

### PR DESCRIPTION
**Pull request by yamilmedina**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-1101" title="AR-1101" target="_blank">SubtaskAR-1101</a>  Opening and closing Self user profile screen animation should be fixed
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [X] contains a reference JIRA issue number like `SQPIT-764`

- The **PR Description**
  - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Solutions 
https://wearezeta.atlassian.net/browse/AR-1101

- This adds the configuration needed to use Google's Accompanist library for animations https://google.github.io/accompanist/navigation-animation/
- Applied slide in from right animation transition for self user profile
- Added the possibility for each screen to opt in for animations, if explicitly declared. In case no animations declared, will fallback to None
- If we need to set a global navigation animation for all screens, we are able to achieve that using default parameters in base class `NavigationItem`

### Attachments (Optional)

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [X] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: app/src/main/kotlin/com/wire/android/navigation/NavigationItem.kt [hunk 1]
```diff
@@ -46,8 +46,8 @@ enum class NavigationItem(
     internal val primaryRoute: String,
     private val canonicalRoute: String,
     val arguments: List<NamedNavArgument> = emptyList(),
-    open val content: @Composable (ContentParams) -> Unit
-    // TODO add animations here
+    open val content: @Composable (ContentParams) -> Unit,
```
Comment (trOnk12) [inline app/src/main/kotlin/com/wire/android/navigation/NavigationItem.kt:49]: why does it have to be open ?

File: app/src/main/kotlin/com/wire/android/navigation/NavigationItem.kt [hunk 2]
```diff
@@ -46,8 +46,8 @@ enum class NavigationItem(
...
+    open val animationConfig: NavigationAnimationConfig = NavigationAnimationConfig.NoAnimationConfig
```
Comment (trOnk12) [inline app/src/main/kotlin/com/wire/android/navigation/NavigationItem.kt:50]: why is this open ?

Comment (yamilmedina) [reply app/src/main/kotlin/com/wire/android/navigation/NavigationItem.kt]: Good point, it was a left over from another approach I was taking before going with this, removed 👍

**Review by trOnk12 — COMMENTED** [review]
looks good to me :) but maybe someone else wants to take a look ?

File: app/src/main/kotlin/com/wire/android/navigation/NavigationAnimationConfig.kt [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+ * Animation that allows a smooth transition from ltr, adding a fade out effect
+ */
+@Suppress("MagicNumber")
+fun smoothSlideOutFromLeft(): ExitTransition {
+    return slideOutHorizontally(animationSpec = spring(stiffness = Spring.StiffnessMediumLow)) {
+        +200
+    } + fadeOut(
+        animationSpec = tween(durationMillis = 200)
+    )
+}
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/navigation/NavigationAnimationConfig.kt:58]: Should we maybe extract to a different `animatedTransitions` file?

File: app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationRouter.kt [hunk 4]
```diff
@@ -104,7 +107,7 @@ private fun ConversationRouter(
         }
 
         with(uiState) {
-            NavHost(conversationState.navHostController, startDestination = ConversationsNavigationItem.All.route) {
+            AnimatedNavHost(conversationState.navHostController, startDestination = ConversationsNavigationItem.All.route) {
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationRouter.kt:110]: Aren't we adding also animated transitions on the Conversation Router?

**Review by gongracr — APPROVED** [review]
Looking much sexier now! 💄 👠

**Review by gongracr — APPROVED** [review]

**Pull request merged**